### PR TITLE
fix(build): Used a fixed temp dir for caching.

### PR DESCRIPTION
**Pull request by petejohanson**
* Improve GHA caching by using a fixed temporary directory when using one.

### CODE REVIEW

**Comment by petejohanson** [comment]
Tested out and caching working using this branch: https://github.com/petejohanson/le-capybara-zmk-config/actions/runs/7906598751/job/21582180373

File: .github/workflows/build-user-config.yml [hunk 1]
```diff
@@ -68,7 +68,8 @@ jobs:
         run: |
           if [ -e zephyr/module.yml ]; then
             export zmk_load_arg=" -DZMK_EXTRA_MODULES='${GITHUB_WORKSPACE}'"
-            export new_tmp_dir=$(mktemp -d)
+            export new_tmp_dir="${TMPDIR:-/tmp}/zmk-config"
```
Comment (caksoylar) [inline .github/workflows/build-user-config.yml:71]: Nit: I guess we don't need to `export` this variable, do we?

**Review by caksoylar — APPROVED** [review]
LGTM, noted a nit.

File: .github/workflows/build-user-config.yml (hunk 1 above)
Comment (petejohanson) [reply .github/workflows/build-user-config.yml]: Nope. Not needed. Fixed, and re-ran the workflow with no error.

**Pull request merged**